File: app/src/main/java/com/kotobyte/models/db/WordMatch.java
package com.kotobyte.models.db;

import java.util.Comparator;

class WordMatch {

    private long mId;
    private String mJson;
    private String mHighlights;
    private float mScore;

    WordMatch(long Id, String Json, String highlights, float score) {
        mId = Id;
        mJson = Json;
        mHighlights = highlights;
        mScore = score;
    }

    long getId() {
        return mId;
    }

    String getJson() {
        return mJson;
    }

    String getHighlights() {
        return mHighlights;
    }

    static class ScoreComparator implements Comparator<WordMatch> {

        @Override
        public int compare(WordMatch o1, WordMatch o2) {
            float scoreDiff = o2.mScore - o1.mScore;

            if (scoreDiff > 0) {
                return -1;
            }

            if (scoreDiff < 0) {
                return 1;
            }

            return 0;
        }
    }
}

File: app/src/main/java/com/kotobyte/models/db/WordMatch.kt
package com.kotobyte.models.db

import java.util.Comparator

internal data class WordMatch(
        val ID: Long,
        val JSON: String,
        val highlights: String,
        private val score: Float
) {

    internal class ScoreComparator : Comparator<WordMatch> {

        override fun compare(o1: WordMatch, o2: WordMatch): Int {
            val scoreDiff = o2.score - o1.score

            if (scoreDiff > 0) {
                return -1
            }

            if (scoreDiff < 0) {
                return 1
            }

            return 0
        }
    }
}
